projects/js-packages/charts/src/components/private/radial-wipe-animation/radial-wipe-animation.tsx

```diff
@@ -0,0 +1,95 @@
+import styles from './radial-wipe-animation.module.scss';
+
+type Angle = `${ number }deg` | `${ number }rad` | `${ number }grad` | `${ number }turn` | 0 | '0';
+
+/**
+ * This animation uses the SVG self-drawing technique (source: https://www.joshwcomeau.com/svg/friendly-introduction-to-svg/),
+ * leveraging stroke-dasharray and stroke-dashoffset properties to create the effect.
+ *
+ * Here, we reverse the animation to "un-draw" the circle for the revealing / radial wipe effect:
+ * - Initially, the entire circle is drawn with a white mask.
+ * - The circle's border (stroke) is drawn with a black stroke, wide enough to cover the circle.
+ * - The stroke is then "un-drawn," creating the wipe animation.
+ * - A white mask makes the area visible, while a black mask makes the area invisible.
+ */
+
+/**
+ * RadialWipeProps - SVG mask props that requires a radial wipe animation effect.
+ *
+ * @param {object}                            props                         - The properties object.
+ * @param {string}                            props.id                      - The unique ID for the mask.
+ * @param {number}                            props.radius                  - The outer radius of the radial wipe.
+ * @param {number}                            [props.innerRadius=0]         - The inner radius of the radial wipe.
+ * @param {number}                            [props.durationMs=1000]       - The duration of the animation in milliseconds.
+ * @param {number}                            [props.top=0]                 - The top offset for the mask.
+ * @param {number}                            [props.left=0]                - The left offset for the mask.
+ * @param {number}                            [props.wipePercentage=100]    - The percentage of the wipe animation to complete.
+ * @param {'clockwise' | 'counter-clockwise'} [props.direction='clockwise'] - The direction of the wipe animation.
+ * @param {Angle}                             [props.startAngle='-90deg']   - The starting angle of the wipe animation.
+ *
+ * @return {JSX.Element} The radial wipe mask element.
+ */
+export type RadialWipeProps = {
+	id: string;
+	radius: number;
+	innerRadius?: number;
+	durationMs?: number;
+	top?: number;
+	left?: number;
+	startAngle?: Angle;
+	direction?: 'clockwise' | 'counter-clockwise';
+	wipePercentage?: number;
+};
+
+/**
+ * Renders a SVG mask that creates a radial wipe animation effect.
+ *
+ * @param {RadialWipeProps} props - Component props
+ * @return {JSX.Element} The rendered mask component
+ */
+function RadialWipe( {
```

Comment:
Related to comment about exports:
```suggestion
export type RadialWipeAnimationProps = {
	id: string;
	radius: number;
	innerRadius?: number;
	durationMs?: number;
	top?: number;
	left?: number;
	startAngle?: Angle;
	direction?: 'clockwise' | 'counter-clockwise';
	wipePercentage?: number;
};

/**
 * Renders a SVG mask that creates a radial wipe animation effect.
 *
 * @param {RadialWipeAnimationProps} props - Component props
 * @return {JSX.Element} The rendered mask component
 */
function RadialWipeAnimation( {
```